### PR TITLE
Add a postprocessor that computes the inner product of two variables

### PR DESCRIPTION
**Pull request by snschune**
Adding a pp computing the inner product of two variables along with a test.

### CODE REVIEW

**Comment by friedmud** [comment]
Assuming you double checked the results of your test analytically... I'll merge this when the testing is finished...

**Comment by snschune** [comment]
What would the default quadrature for `Second Monomial` be? The integration 
for `P_2` should give `0.4` and it gives `0.3996`. That looks like a truncation error.
I thought it's good enough.

**Comment by friedmud** [comment]
I don't know - you could try jacking up the integration order though...

Check out the `Quadrature` block here:

https://github.com/idaholab/moose/blob/devel/test/tests/quadrature/order/order5.i

**Comment by snschune** [comment]
I'll do that, wait with the merging till then.

File: test/tests/postprocessors/variable_inner_product/gold/variable_inner_product.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+time,f_dot_f,f_dot_g,norm_f
+0,0,0,0
+1,0.4,4.1633363423443e-17,0.63245553203368
```
Comment (snschune) [inline test/tests/postprocessors/variable_inner_product/gold/variable_inner_product.csv]: 0.4 is the answer I wanted to see.

Comment (permcody) [inline test/tests/postprocessors/variable_inner_product/gold/variable_inner_product.csv]: I don't know, it's a fraction. I like "1" better! 😄

**Comment by friedmud** [comment]
Ok - looks good.  When the tests pass I'll merge.

File: test/tests/postprocessors/variable_inner_product/gold/variable_inner_product.csv (hunk 1 above)
Comment (snschune) [inline test/tests/postprocessors/variable_inner_product/gold/variable_inner_product.csv]: These are not orthonormal Legendre polynomials, just orthogonal so `P_n * P_m = delta_{n,m} 2 / (2*n+1)`.

**Comment by friedmud** [comment]
Sorry to leave this hanging...

**Pull request merged**